posts/2024-03-29-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,161 @@
+
+[#deploy_app_with_gitops]
+== Deploying a Java application with GitOps and ArgoCD
+
+Now that you have the Git repository forked and ready for deployment, it’s time to configure ArgoCD to deploy the application.
+
+First, go to the ArgoCD console dashboard that we viewed before.
+
+Next, click the New App button near the top left of the console.  This opens a window where you will give ArgoCD the details on the application you want to deploy, which you can see an example of in the image below.
+
+For our demo application, we've called the application name "daytrader7". We've checked the Auto-Create Namespace box which creates the namespace if it doesn't already exist.  We've left the ArgoCD project as "default" as this is a way to group applications together for large scale deployments, which is not needed in our case. And, we've set the Sync Policy to Manual. This configuration option determines whether we want ArgoCD to automatically correct configuration drift or wait for us to manually tell it to.  
+[.img_border_light]
+image::/img/blog/configuration-drift-3.png[Creating the App,width=70%,align="center"]
+
+Select the configiration options that best suit your application. For the purposes of just trying out and getting started with ArgoCD, feel free to copy the configuration we've set in our example app.
+
+Next, you will need to scroll down to the configuration section labelled "Source", shown in the image below. Here, you will need to set several configuration options.
+
+First, change the Repository URL to your newly forked repo. This tells ArgoCD what git repository has the configuration for the application you are looking to deploy. 
+
+You'll also need to change the path, which is the path in the git repo where ArgoCD is looking for deployment files to deploy. 
+
+Additionally, you'll need to set the Cluster URL. In this example, we have set it to https://kubernetes.default.svc, which is the local cluster URL.
+
+Then, lastly, you will need to set the namespace. In our case, this is daytrader7 (which will be created because we selected the "Auto-Create Namespace" option).
+
+
+After verifying these options, click Create in the top left.  This creates an "Application" resource that ArgoCD uses to watch the repository we gave it.  
+
+[.img_border_light]
+image::/img/blog/configuration-drift-4.png[Setting the Repo,width=70%,align="center"]
+
+[#manually_sync]
+=== Manually Syncing your Application with the Repository
+
+Since we chose manual sync policy in the options for demonstrative purposes, we have to sync the application manually for ArgoCD to deploy the application. If you have also selected this option, on the main page of the ArgoCD console, you should see a new tile that looks like this:
```

Comment:
```suggestion
Since we chose a manual sync policy, we have to sync the application manually for ArgoCD to deploy the it. If you have also selected this option, on the main page of the ArgoCD console, you should see a new tile that looks like this:
```